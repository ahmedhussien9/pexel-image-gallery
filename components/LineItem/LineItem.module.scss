.article {
    cursor: pointer;
    display: flex;
    flex-direction: column;
    align-items: center;
    .link {
        width: 100%;
        height: 100%;
        display: block;
        position: relative;
        text-align: center;
        position: relative;
        background-color: transparent;
        cursor: pointer;
        border: 0;
        padding: 0;
        font: inherit;
        .title {
            margin-top: 5px;
        }
        .image {
            transition: all 0.3s ease-in-out;
            &:hover {
                transform: scale(1.2);
            }
        }
    }
}
