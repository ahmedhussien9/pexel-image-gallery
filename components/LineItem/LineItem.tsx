import { IPhoto } from '../../interfaces/photo.interface';
import Image from "next/image";
import CardConainer from '../CardContainer/CardConainer';
import styles from "./LineItem.module.scss";

type Props = {
    item: IPhoto,
    onClickItem: (item) => void;
}

const LineItem = ({ item, onClickItem }: Props) => (
    <article className={styles.article} role="contentinfo" data-meta-title={item.photographer}>
        <button className={styles.link} onClick={() => onClickItem(item)} aria-label="Open the photo in a popup" title="open a popup to see a large image"  >
            <Image aria-label={`${item.alt}`} className={styles.image} src={item.src.medium} alt={item.alt} height={200} width={250} />
            <p aria-label={`The name of photo owner is ${item.photographer}`} className={styles.title} >{item.photographer}</p>
        </button>
    </article>
)

export default LineItem
