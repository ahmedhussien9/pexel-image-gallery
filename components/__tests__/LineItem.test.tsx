import { shallow } from 'enzyme';
import sinon from 'sinon';

import { render, screen } from '@testing-library/react';
import { IPhoto } from '../../interfaces/photo.interface';
import LineItem from "../LineItem/LineItem";

describe('Line item Component', () => {
    let clickedItem: IPhoto = null;
    const item: IPhoto = {
        "id": 2014422,
        "width": 3024,
        "height": 3024,
        "url": "https://www.pexels.com/photo/brown-rocks-during-golden-hour-2014422/",
        "photographer": "Joey Farina",
        "photographer_url": "https://www.pexels.com/@joey",
        "photographer_id": 680589,
        "avg_color": "#978E82",
        "src": {
            "original": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg",
            "large2x": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940",
            "large": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&h=650&w=940",
            "medium": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&h=350",
            "small": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&h=130",
            "portrait": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800",
            "landscape": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200",
            "tiny": "https://images.pexels.com/photos/2014422/pexels-photo-2014422.jpeg?auto=compress&cs=tinysrgb&dpr=1&fit=crop&h=200&w=280"
        },
        "liked": false,
        "alt": "Brown Rocks During Golden Hour"
    };

    const getClickedItem = (item: IPhoto) => {
        clickedItem = item;
    }

    it('Should render a snapshot of the line item component', () => {
        const { container } = render(<LineItem item={item} onClickItem={null} />);
        expect(container).toMatchSnapshot();
    });

    it('Should render a title with correct title', () => {
        const wrapped = shallow(<LineItem item={item} onClickItem={null} />);
        expect(wrapped.find('.title').text()).toEqual(item.photographer)
    });

    it('Should test the click on photo item', () => {
        const onClickItem = sinon.spy();
        const wrapper = shallow(<LineItem item={item} onClickItem={() => onClickItem(item)} />);
        wrapper.find('button').simulate('click');
        expect(onClickItem).toHaveProperty('callCount', 1);
    });

    it('Should get the clicked item', () => {
        const wrapped = shallow(<LineItem item={item} onClickItem={() => getClickedItem(item)} />);
        const button = wrapped.find('button').simulate('click');
        expect(clickedItem.id).toEqual(2014422);
        expect(clickedItem.alt).toBe('Brown Rocks During Golden Hour');
    });

})