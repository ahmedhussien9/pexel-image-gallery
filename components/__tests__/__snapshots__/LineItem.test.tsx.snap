// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line item Component Should render a snapshot of the line item component 1`] = `
<div>
  <article
    class="article"
    data-meta-title="Joey Farina"
    role="contentinfo"
  >
    <button
      aria-label="Open the photo in a popup"
      class="link"
      title="open a popup to see a large image"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27250%27%20height=%27200%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Brown Rocks During Golden Hour"
          aria-label="Brown Rocks During Golden Hour"
          class="image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p
        aria-label="The name of photo owner is Joey Farina"
        class="title"
      >
        Joey Farina
      </p>
    </button>
  </article>
</div>
`;
